include/BSMPT/utility/utility.h

```diff
@@ -67,6 +108,144 @@ std::ostream &operator<<(std::ostream &os, const std::vector<T> &vec)
   return os;
 }
 
+/**
+ * @brief vector addition
+ */
+template <typename T>
+std::vector<T> operator+(const std::vector<T> &a, const std::vector<T> &b)
+{
+  if (a.size() != b.size())
+    throw("Vector cannot be added. Must have the same size.");
```

Comment:
```suggestion
    throw std::runtime_error("Vector cannot be added. Must have the same size.");
```
Also for the other throws, you need an exception type